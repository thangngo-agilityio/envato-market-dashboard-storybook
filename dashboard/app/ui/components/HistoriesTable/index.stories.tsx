import { StoryObj, Meta } from '@storybook/react';
import { Box } from '@chakra-ui/react';

// Component
import HistoriesTable from '@/ui/components/HistoriesTable';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient();

const meta: Meta<typeof HistoriesTable> = {
  title: 'Custom Components/HistoriesTable',
  tags: ['autodocs'],
  component: HistoriesTable,
  decorators: [
    (Story) => (
      <QueryClientProvider client={queryClient}>
        <Box
          bgColor="background.component.primary"
          borderRadius={8}
          px={6}
          py={5}
        >
          <Story />
        </Box>
      </QueryClientProvider>
    ),
  ],
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;
type Story = StoryObj<typeof HistoriesTable>;

export const Default: Story = {
  args: {},
};
