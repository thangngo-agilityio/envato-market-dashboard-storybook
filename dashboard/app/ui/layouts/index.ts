export { default as SideBar } from './Sidebar';
export { default as Header } from './Header';
export { default as AuthHeader } from './AuthHeader';
export { default as AuthFooter } from './AuthFooter';
export { default as MainLayout } from './MainLayout';
