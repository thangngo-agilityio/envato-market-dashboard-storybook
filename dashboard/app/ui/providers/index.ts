export { default as ChakraProvider } from './ChakraUIProvider';
export { default as QueryProvider } from './QueryProvider';
export { default as CheckAuthenticationProvider } from './CheckAuthenticationProvider';
export { default as CheckPinCodeProvider } from './CheckPinCodeProvider';
