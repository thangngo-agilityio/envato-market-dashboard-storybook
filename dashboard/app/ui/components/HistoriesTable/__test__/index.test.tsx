import { useSearchParams, useRouter } from 'next/navigation';
import axios from 'axios';

// Components
import HistoriesTable from '..';

// Hooks
import * as hooks from '@/lib/hooks';

// Provider
import { QueryProvider } from '@/ui/providers';

// Mocks
import { TRANSACTIONS } from '@/lib/mocks';

// Constants
import { DEBOUNCE_TIME } from '@/lib/constants';

jest.mock('@/lib/hooks', () => ({
  ...jest.requireActual('@/lib/hooks'),
  usePagination: jest.fn(),
  useTransactions: jest.fn(),
}));

jest.mock('axios');

const resetMock = jest.fn();
const sortByMock = jest.fn();
const deleteTransactionMock = jest.fn();
const updateTransactionMock = jest.fn();

const SORT_ICON = 'sort-icon';
const SEARCH_TRANSACTION = 'search-transaction';

const setup = () =>
  render(<HistoriesTable />, {
    wrapper: QueryProvider,
  });

describe('Histories table', () => {
  beforeEach(() => {
    jest.spyOn(hooks, 'usePagination').mockReturnValue({
      resetPage: resetMock,
      filterData: TRANSACTIONS,
      data: [1],
    } as unknown as ReturnType<typeof hooks.usePagination>);
    jest.spyOn(axios, 'get').mockResolvedValue(TRANSACTIONS);
    (useSearchParams as jest.Mock).mockReturnValue({
      get: jest.fn(),
      forEach: jest.fn(),
    });
    (useRouter as jest.Mock).mockReturnValue({
      push: jest.fn(),
    });
    jest.spyOn(hooks, 'useTransactions').mockReturnValue({
      isLoading: false,
      dataHistory: TRANSACTIONS,
      dataTransaction: TRANSACTIONS,
      sortBy: sortByMock,
      deleteTransaction: deleteTransactionMock,
      updateTransaction: updateTransactionMock,
    } as unknown as ReturnType<typeof hooks.useTransactions>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Match with snapshot with default prop', () => {
    const { container } = setup();

    waitFor(() => expect(container).toMatchSnapshot());
  });

  jest.useFakeTimers();
  it('Typing input search', async () => {
    const { getByTestId } = setup();
    const input = getByTestId(SEARCH_TRANSACTION) as HTMLInputElement;

    await act(async () => {
      fireEvent.change(input, {
        target: {
          value: '123',
        },
      });
    });

    jest.advanceTimersByTime(DEBOUNCE_TIME);
    expect(resetMock).toHaveBeenCalled();
  });

  it('Sort transaction', async () => {
    const { getAllByTestId } = setup();
    const sortIcon = getAllByTestId(SORT_ICON);

    await act(async () => {
      fireEvent.click(sortIcon[0]);
    });

    expect(sortByMock).toHaveBeenCalled();
  });
});
