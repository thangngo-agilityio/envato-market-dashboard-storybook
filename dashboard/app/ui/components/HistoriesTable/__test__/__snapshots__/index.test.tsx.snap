// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Histories table Match with snapshot with default prop 1`] = `
<div>
  <form
    class="chakra-stack css-16wym0o"
    data-testid="search-bar"
  >
    <div
      class="css-1b9rej7"
    >
      <div
        class="chakra-form-control css-1kxonj9"
        role="group"
      >
        <div
          class="chakra-input__group css-1091msq"
          data-group="true"
        >
          <div
            class="chakra-input__left-element css-1kg931p"
          >
            <svg
              class="chakra-icon css-onkibi"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.414,20.591l-4.645-4.645a10.256,10.256,0,1,0-2.828,2.829l4.645,4.644a2.025,2.025,0,0,0,2.828,0A2,2,0,0,0,23.414,20.591ZM10.25,3.005A7.25,7.25,0,1,1,3,10.255,7.258,7.258,0,0,1,10.25,3.005Z"
                fill="currentColor"
              />
            </svg>
          </div>
          <input
            class="chakra-input css-1ybypo"
            data-testid="search-transaction"
            id="field-:r0:"
            placeholder="Search by name, email, or other..."
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="css-tzs3vb"
    >
      <button
        aria-controls="menu-list-:r2:"
        aria-expanded="false"
        aria-haspopup="menu"
        class="chakra-button chakra-menu__menu-button css-1d0mwls"
        id="menu-button-:r2:"
        type="button"
      >
        <span
          class="css-xl71ch"
        >
          <span
            class="css-ccj5l2"
          >
            <svg
              fill="none"
              height="17"
              stroke="#1D1E24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.552 13.502H1.25M10.362 3.81h6.301"
              />
              <path
                clip-rule="evenodd"
                d="M5.948 3.756a2.355 2.355 0 0 0-2.363-2.347 2.355 2.355 0 0 0-2.363 2.347 2.355 2.355 0 0 0 2.363 2.346 2.355 2.355 0 0 0 2.363-2.346ZM17.221 13.463a2.354 2.354 0 0 0-2.362-2.346 2.355 2.355 0 0 0-2.364 2.346 2.355 2.355 0 0 0 2.364 2.347 2.354 2.354 0 0 0 2.362-2.347Z"
              />
            </svg>
            Filters
          </span>
        </span>
      </button>
      <div
        class="css-r6z5ec"
        style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-1k4isgz"
          id="menu-list-:r2:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="0"
            id="menu-list-:r2:-menuitem-:r3:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              January
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="1"
            id="menu-list-:r2:-menuitem-:r4:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              February
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="2"
            id="menu-list-:r2:-menuitem-:r5:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              March
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="3"
            id="menu-list-:r2:-menuitem-:r6:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              April
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="4"
            id="menu-list-:r2:-menuitem-:r7:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              May
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="5"
            id="menu-list-:r2:-menuitem-:r8:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              June
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="6"
            id="menu-list-:r2:-menuitem-:r9:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              July
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="7"
            id="menu-list-:r2:-menuitem-:ra:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              August
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="8"
            id="menu-list-:r2:-menuitem-:rb:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              September
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="9"
            id="menu-list-:r2:-menuitem-:rc:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              October
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="10"
            id="menu-list-:r2:-menuitem-:rd:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              November
            </p>
          </button>
          <button
            class="chakra-menu__menuitem css-o9weuj"
            data-index="11"
            id="menu-list-:r2:-menuitem-:re:"
            role="menuitem"
            tabindex="-1"
            type="button"
          >
            <p
              class="chakra-text css-15asmb0"
            >
              December
            </p>
          </button>
        </div>
      </div>
    </div>
  </form>
  <div
    class="css-17qnj1m"
  >
    <div
      class="chakra-table__container css-1d9a4hd"
    >
      <table
        class="chakra-table css-5605sr"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
          >
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Customer name"
                >
                  Customer name
                </p>
                <button
                  aria-label="This is the icon for Customer name"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Date"
                >
                  Date
                </p>
                <button
                  aria-label="This is the icon for Date"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Amount"
                >
                  Amount
                </p>
                <button
                  aria-label="This is the icon for Amount"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Payment"
                >
                  Payment
                </p>
                <button
                  aria-label="This is the icon for Payment"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
            <th
              class="css-1fgb9gn"
            >
              <div
                class="css-1k9efnl"
              >
                <p
                  class="chakra-text css-10ze6i3"
                  title="Status"
                >
                  Status
                </p>
                <button
                  aria-label="This is the icon for Status"
                  class="chakra-button css-1c8sy4r"
                  data-testid="sort-icon"
                  type="button"
                >
                  <svg
                    fill="none"
                    height="12"
                    viewBox="0 0 14 12"
                    width="14"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                    />
                    <path
                      clip-rule="evenodd"
                      d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                      fill="#969BA0"
                      fill-rule="evenodd"
                      opacity="1"
                    />
                  </svg>
                </button>
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
        >
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1mu11wg"
            >
              <div
                class="css-12agq4v"
              >
                <div
                  class="css-3mb4de"
                >
                  <img
                    alt="Image of Devon Lane"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                  />
                </div>
                <p
                  class="chakra-text css-b5wh8g"
                >
                  Devon Lane
                </p>
              </div>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                Feb 26, 1689
              </p>
            </td>
            <td
              class="css-5x3kvh"
            >
              <p
                class="chakra-text css-1mlmrnl"
              >
                $101.00
              </p>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-1nt00i9"
              >
                Paid
              </span>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-19mxxmq"
              >
                Cancelled
              </span>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1mu11wg"
            >
              <div
                class="css-12agq4v"
              >
                <div
                  class="css-3mb4de"
                >
                  <img
                    alt="Image of Bessie Cooper"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Ff_auto%2Cq_auto%3Aeco%2Faf53d53d-561f-450a-a483-70a7ceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                  />
                </div>
                <p
                  class="chakra-text css-b5wh8g"
                >
                  Bessie Cooper
                </p>
              </div>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                Feb 26, 1689
              </p>
            </td>
            <td
              class="css-5x3kvh"
            >
              <p
                class="chakra-text css-1mlmrnl"
              >
                $101.00
              </p>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-1nt00i9"
              >
                Paid
              </span>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-1nt00i9"
              >
                Completed
              </span>
            </td>
          </tr>
          <tr
            class="css-0"
            data-testid="table-row"
          >
            <td
              class="css-1mu11wg"
            >
              <div
                class="css-12agq4v"
              >
                <div
                  class="css-3mb4de"
                >
                  <img
                    alt="Image of Zepeda lorna"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    sizes="100vw"
                    src="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75"
                    srcset="/_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fstatic.nike.com%2Fa%2Fimages%2Ft_PDP_1280_v1%2Fceee380f%2Fdunk-low-shoes-t9dFBx.png&w=3840&q=75 3840w"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent; border-radius: 50%; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                  />
                </div>
                <p
                  class="chakra-text css-b5wh8g"
                >
                  Zepeda lorna
                </p>
              </div>
            </td>
            <td
              class="css-1v6q597"
            >
              <p
                class="chakra-text css-gmde4n"
              >
                Feb 26, 1689
              </p>
            </td>
            <td
              class="css-5x3kvh"
            >
              <p
                class="chakra-text css-1mlmrnl"
              >
                $101.00
              </p>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-oyldzo"
              >
                Unpaid
              </span>
            </td>
            <td
              class="css-3qub4l"
            >
              <span
                class="chakra-badge css-1aary8z"
              >
                Pending
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
